homeassistant/components/hive/sensor.py

```diff
@@ -1,20 +1,100 @@
 """Support for the Hive sensors."""
+from dataclasses import dataclass
 from datetime import timedelta
 
-from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
 from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import TEMP_CELSIUS
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from . import HiveEntity
 from .const import DOMAIN
 
 PARALLEL_UPDATES = 0
 SCAN_INTERVAL = timedelta(seconds=15)
-DEVICETYPE = {
-    "Battery": {"unit": " % ", "type": SensorDeviceClass.BATTERY},
-}
+
+
+@dataclass
+class HiveSensorEntityDescription(SensorEntityDescription):
```

Comment:
I'd we don't need any custom attributes we can use `SensorEntityDescription` directly.